add support for namePrefix option, to avoid conflicts with generated names

kevingessner opened the pull request:
My particular use case of this is having two dependencies.yaml files in the same repository, with different versions of certain artifacts and different `thirdPartyDirectory`s.  Without this prefixing, the names used in the workspace and for `bind` conflict.

johnynek commented on README.md:
```diff
@@ -105,13 +105,15 @@ In the options we set:
 * resolverCache: where bazel-deps should cache resolved packages.  `local` (`target/local-repo` in the repository root)
   or `bazel_output_base` (`bazel-deps/local-repo` inside the repository's Bazel output base -- from `bazel info
   output_base`)
+* namePrefix: a string appended to the generated workspace names, to avoid conflicts
```
can we add an example of this? I think it just changes the binding name?

kevingessner commented:
@johnynek Added a longer comment and a test with some examples.

johnynek commented:
looks like the tests fail.

Did they pass for you locally?

kevingessner commented:
Updated to fix the codacy error, and rebased onto master.

Tests pass locally here.  I re-ran parsetest a few times in case it was transient but I couldn't repro.  Is there some trick to debugging tests with Gen, with the randomness and all?

kevingessner commented:
thanks!
